diff --git mlibc-clean/sysdeps/linux/generic/sysdeps.cpp mlibc-workdir/sysdeps/linux/generic/sysdeps.cpp
index 34a5690..f1986a7 100644
--- mlibc-clean/sysdeps/linux/generic/sysdeps.cpp
+++ mlibc-workdir/sysdeps/linux/generic/sysdeps.cpp
@@ -1510,6 +1510,30 @@ int sys_setschedparam(void *tcb, int policy, const struct sched_param *param) {
 	return 0;
 }
 
+int sys_getscheduler(pid_t pid, int *sched) {
+	auto ret = do_syscall(SYS_sched_getscheduler, pid);
+	if (int e = sc_error(ret); e)
+		return e;
+	*sched = sc_int_result<int>(ret);
+	return 0;
+}
+
+int sys_sched_get_priority_max(int policy, int *out) {
+	auto ret = do_syscall(SYS_sched_get_priority_max, policy);
+	if (int e = sc_error(ret); e)
+		return e;
+	*out = sc_int_result<int>(ret);
+	return 0;
+}
+
+int sys_sched_get_priority_min(int policy, int *out) {
+	auto ret = do_syscall(SYS_sched_get_priority_min, policy);
+	if (int e = sc_error(ret); e)
+		return e;
+	*out = sc_int_result<int>(ret);
+	return 0;
+}
+
 int sys_getparam(pid_t pid, struct sched_param *param) {
 	auto ret = do_syscall(SYS_sched_getparam, pid, param);
 	if (int e = sc_error(ret); e)
@@ -2098,6 +2122,13 @@ int sys_sigsuspend(const sigset_t *set) {
 	return 0;
 }
 
+int sys_sigpending(sigset_t *set) {
+	auto ret = do_syscall(SYS_rt_sigpending, set, NSIG / 8);
+	if (int e = sc_error(ret); e)
+		return e;
+	return 0;
+}
+
 int sys_sigaltstack(const stack_t *ss, stack_t *oss) {
 	auto ret = do_syscall(SYS_sigaltstack, ss, oss);
 	if (int e = sc_error(ret); e)
@@ -2204,6 +2235,14 @@ int sys_readlink(const char *path, void *buf, size_t bufsiz, ssize_t *len) {
 	return 0;
 }
 
+int sys_readlinkat(int dirfd, const char *path, void *buffer, size_t max_size, ssize_t *length) {
+	auto ret = do_syscall(SYS_readlinkat, dirfd, path, buffer, max_size);
+	if (int e = sc_error(ret); e)
+		return e;
+	*length = sc_int_result<ssize_t>(ret);
+	return 0;
+}
+
 #if __MLIBC_BSD_OPTION
 // getloadavg() adapted from musl
 int sys_getloadavg(double *samples) {
